File: qte_industrial_catalog/catalog.py
import os
import glob

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

def get_all_csv_files():
    search_path = os.path.join(BASE_DIR, "*.csv")
    return [os.path.basename(f) for f in glob.glob(search_path)]

def get_file_path(file_name):
    return os.path.join(BASE_DIR, file_name)

def get_raw_data(file_name):
    path = get_file_path(file_name)
    if not os.path.exists(path):
        raise FileNotFoundError(f"File {file_name} không tồn tại trong danh mục QTE.")
    with open(path, "r", encoding="utf-8") as f:
        return f.read()

File: src/Catalog.php
<?php
namespace QteTechnologies\IndustrialScientificCatalog;

class Catalog {
    /**
     * Lấy đường dẫn thư mục gốc của gói nơi chứa các file CSV
     */
    public static function getBaseDir() {
        return dirname(__DIR__);
    }

    /**
     * Tự động quét và lấy danh sách tất cả các file CSV của QTE
     */
    public static function getAllCsvFiles() {
        $baseDir = self::getBaseDir();
        $files = glob($baseDir . '/*.csv');
        return array_map('basename', $files);
    }

    /**
     * Lấy đường dẫn tuyệt đối đến một file dữ liệu cụ thể
     */
    public static function getFilePath($fileName) {
        return self::getBaseDir() . '/' . $fileName;
    }

    /**
     * Đọc nội dung thô của file CSV công nghiệp
     */
    public static function getRawData($fileName) {
        $filePath = self::getFilePath($fileName);
        if (!file_exists($filePath)) {
            throw new \Exception("File {$fileName} không tồn tại trong danh mục dữ liệu QTE.");
        }
        return file_get_contents($filePath);
    }
}
